### DISCUSSION TITLE
ABCDE Configuration

### DISCUSSION THREAD

**Discussion by trleith** [automatic-ripping-machine/automatic-ripping-machine#798] [General]
I'm back to fooling with ARM again, this time for ripping CDs

I've wanted to make the file naming patterns work for 1)Windows & Unix both, and 2) Plex

I've never heard of recode (which seems it might be superseded by iconv, but nevermind for now). The sample abcde.conf contains the following:

```
# VARIANT 2: (legible style)
#
# Accept all chars, EXCEPT '/' (obvious) or ":" (because eyeD3 cannot
# cope with ":" in pathnames): replace them with " " (space).
# Additionally, replace contiguous spaces with one space; strip
# leading spaces; strip trailing spaces and recode to flat filenames.
#
# mungefilename ()
# {
#       echo "$@" |     sed "s/[:\/]/ /g" | \
#                       sed 's/ [ ]*/ /g' | \
#                       sed 's/^ *//' | \
#                       sed 's/ *$//' | \
#                       recode -f iso8859-1..flat
```
recode is not included in the automaticrippingmachine/automatic-ripping-machine:nvidia Docker image
Instead use iconv -f iso8859-1 -t ASCII//TRANSLIT
If there's an installation script for ARM in Docker, it might be worthwhile making this substitution on the way in to avoid support headaches. iconv might work better than recode anyway.

=============== Next Problem ===================
If you edit the abcde.conf file in the ARM web UI using Edge, it gets \r [0x0d] [0o013] appended to at least some of the empty lines, which makes abcde or an ARM script (not sure which) barf. You will see something like this:

```
/etc/arm/config/abcde.conf: line 8: $'\r': command not found
/etc/arm/config/abcde.conf: line 14: $'\r': command not found
/etc/arm/config/abcde.conf: line 19: $'\r': command not found
/etc/arm/config/abcde.conf: line 23: $'\r': command not found
/etc/arm/config/abcde.conf: line 28: $'\r': command not found
/etc/arm/config/abcde.conf: line 31: $'\r': command not found
/etc/arm/config/abcde.conf: line 37: $'\r': command not found
/etc/arm/config/abcde.conf: line 41: $'\r': command not found
/etc/arm/config/abcde.conf: ... **lots of them**
```

This can be fixed by:
sudo -i
docker cp arm-rippers/etc/.abcde.conf ./abcde.conf.borked
tr -d "\r" "" << ./abcde.conf.borked > ,/abcde.conf
< if you have further edits to make, use vi and make them here>
docker cp ./abcde.conf arm-rippers:/etc/.abcde.conf
docker restart arm-rippers

For now, don't edit abcde.conf with the UI.

**Comment by microtechno9000** [comment]
Hi @trleith , the issue with the arm config should have been resolved with a recent PR.
If you ar running the latest ARM, can you confirm this issue is no longer present.

**Reply by trleith** [reply]
Thanks.

For now I'm using automaticrippingmachine/automatic-ripping-machine:nvidia

but when the nvidia branch gets merged back into the main line, I shall check.

**Reply by 1337-server** [reply]
> Thanks.
> 
> 
> 
> For now I'm using automaticrippingmachine/automatic-ripping-machine:nvidia
> 
> 
> 
> but when the nvidia branch gets merged back into the main line, I shall check.

The nvidia branch has already been merged into main.

**Reply by trleith** [reply]
Terrific. I'll make the switch
